### ISSUE TITLE
Split INFO fields

### ISSUE THREAD

**Issue by dakl** [moonso/vcf_parser#4]
Hi, 

TL;DR: vcf_parser doesn't split the INFO field when there are multiple values. 

The info field in a VCF can have multiple values for a given attribute. Example: 

```
##INFO=<ID=CNT,Number=A,Type=Integer,Description="Number of times this allele was found in external db">
```

This adds a field named CNT, with a single value for each ALT allele (Number=A, see VCF spec for others variants of this such as R = one per allele including ref allele, new in VCF4.2). To test this, I modified test_vcf.vcf in the examples folder to look like this: 

```
##fileformat=VCFv4.1
##INFO=<ID=MQ,Number=1,Type=Float,Description="RMS Mapping Quality">
##INFO=<ID=CNT,Number=A,Type=Integer,Description="Number of times this allele was found in external db">
##contig=<ID=1,length=249250621,assembly=b37>
##reference=file:///humgen/gsa-hpprojects/GATK/bundle/current/b37/human_g1k_v37.fasta
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	father	mother	proband
1	11900	.	A	T	100	PASS	MQ=1	GT:GQ	0/1:60	0/1:60	1/1:60
1	879585	.	A	T	100	PASS	MQ=1	GT:GQ	0/1:60	0/0:60	0/1:60
1	879586	.	A	T	100	PASS	MQ=1	GT:GQ	0/0:60	0/1:60	0/1:60
1	947378	.	A	T	100	PASS	MQ=1	GT:GQ	0/0:60	0/0:60	0/1:60
1	973348	.	G	A	100	PASS	MQ=1	GT:GQ	0/0:60	0/0:60	0/1:60
3	879585	.	A	T	100	PASS	MQ=1	GT:GQ	0/1:60	0/0:60	0/1:60
3	879586	.	A	T	100	PASS	MQ=1	GT:GQ	0/0:60	0/1:60	0/1:60
3	947378	.	A	T	100	PASS	MQ=1	GT:GQ:AD:DP	0/0:60:5,7:12	0/0:60:4,6:14	0/1:60:7,8:16
3	947379	.	A	T,C	100	PASS	MQ=1;CNT=5,8	GT:GQ:AD:DP	1/1:60:0,7,0:12	0/2:60:7,0,10:17	1/2:60:0,7,8:16
3	973348	.	G	A	100	PASS	MQ=1;CNT=12	GT:GQ:TP	0/0:60:23	0/0:60	0/1:60
3	973349	.	G	A,T	100	PASS	MQ=1;CNT=7,3	GT:GQ:TP	0/1:60:23	1/2:60	0/2:60
```

The three last variants have the CNT attribute in the INFO field, with one value per ALT allele. However, running `vcf_parser examples/test_vcf.vcf --split` doesn't split these correctly, or at all. The last three variants (giving 5 rows after the split) look like this: 

```
3	947379	.	A	T	100	PASS	MQ=1;CNT=5,8	GT:GQ:AD:DP	1/1:60:0,7:12	0/0:60:7,0:17	0/1:60:0,7:16
3	947379	.	A	C	100	PASS	MQ=1;CNT=5,8	GT:GQ:AD:DP	0/0:60:0,0:12	0/1:60:7,10:17	0/1:60:0,8:16
3	973348	.	G	A	100	PASS	MQ=1;CNT=12	GT:GQ:TP	0/0:60:23	0/0:60	0/1:60
3	973349	.	G	A	100	PASS	MQ=1;CNT=7,3	GT:GQ:TP	0/1:60:23	0/1:60	0/0:60
3	973349	.	G	T	100	PASS	MQ=1;CNT=7,3	GT:GQ:TP	0/0:60:23	0/1:60	0/1:60
```

The CNT attribute is just passed along to each of the new lines, thereby dropping the association from allele to value. I think they should be split so that a single value remains on each of the new lines. (Two values for attributes that are Number=R in the header).

cheers

Daniel

**Comment by moonso** [comment]
Thanks.
Yes this should defenitely be the case, I thought it would work...
I'll check this and come back!

**Comment by dakl** [comment]
Great, thanks!

**Comment by moonso** [comment]
Fixed this in version 1.1.7.
I've also added your example to examples/test_vcf.vcf :)
Please try it.

**Comment by dakl** [comment]
```

3 947379 . A T 100 PASS MQ=1;CNT=5 GT:GQ:AD:DP 1/1:60:0,7:12 0/0:60:7,0:17
0/1:60:0,7:16

3 947379 . A C 100 PASS MQ=1;CNT=8 GT:GQ:AD:DP 0/0:60:0,0:12 0/1:60:7,10:17
0/1:60:0,8:16

3 973348 . G A 100 PASS MQ=1;CNT=12 GT:GQ:TP 0/0:60:23 0/0:60 0/1:60

3 973349 . G A 100 PASS MQ=1;CNT=7 GT:GQ:TP 0/1:60:23 0/1:60 0/0:60

3 973349 . G T 100 PASS MQ=1;CNT=3 GT:GQ:TP 0/0:60:23 0/1:60 0/1:60
```

Works great, thanks! =)

**Comment by moonso** [comment]
Nice

**Comment by dakl** [comment]
Tried mailing this, but it seems not to have ended up in this thread so pasting: 

On a side note, vcf_parser currently supports integers, ".", A and G count types. R (one value per allele incl ref) is new in the VCF 4.2 spec. 

From the current header parsing:
```
self.info_pattern = re.compile(r'''\#\#INFO=<
            ID=(?P<id>[^,]+),
            Number=(?P<number>-?\d+|\.|[AG]),
            Type=(?P<type>Integer|Float|Flag|Character|String),
            Description="(?P<desc>[^"]*)"
            >''', re.VERBOSE)
```

For example, in the next version of ExAC, the attributes DP_HIST and GQ_HIST will be of type Number=R. Those fields are currently (incorrectly) A, see https://github.com/konradjk/exac_browser/issues/164. 

/D

**Comment by moonso** [comment]
Thank's alot for this Daniel.
I've updated the parser and added one example(from the thread you pointed to) to test_vcf.vcf in version 1.1.8.

/Måns
